rockcraft.yaml

```diff
@@ -87,9 +89,51 @@ parts:
     stage:
       - opt/spark/jars
 
-  user-setup:
+  client-scripts:
     plugin: nil
     after: [hadoop-jars]
+    build-packages:
+      - wget
+    overlay-packages:
+      - python3-pip
+    overlay-script: |
+      SPARK_CLIENT_PKG_VERSION='0.0.2'
+      mkdir -p $CRAFT_PART_INSTALL/opt/spark/python/dist
+      cd $CRAFT_PART_INSTALL/opt/spark/python/dist
+      echo "Downloading latest available Spark wheel package release v${SPARK_CLIENT_PKG_VERSION}."
+      wget https://github.com/canonical/spark-client-snap/releases/download/v${SPARK_CLIENT_PKG_VERSION}/spark_client-${SPARK_CLIENT_PKG_VERSION}-py3-none-any.whl --no-check-certificate
```

Comment:
`--no-check-certificate` man-in-the-middle attack?
AFAIK github has a trusted certs, maybe install `ssl-cert` together with `wget`?

Reply:
The code block:
```
      wget ...
      if  [[ $? -ne 0 ]]
        then
```
can be one liner.

```
if ! wget ... ; then
  echo "Error: ..." >&2
  exit 1
fi
```

It will allow you to enable "set -e" in the future => improve errors handling.